weasyl/controllers/moderation.py

```diff
@@ -219,3 +219,201 @@ def modcontrol_copynotetostaffnotes_post_(request):
         message=notedata['content'],
     )
     raise HTTPSeeOther("/staffnotes/" + notedata['sendername'])
+
+
+@moderator_only
+def modcontrol_spamqueue_journal_get_(request):
+    """
+    A Pyramid route controller for displaying any journal items which the spam filter has tagged as being spam.
+    :param request: The Pyramid request.
+    :return: A rendered page for the journal spam queue.
+    """
+    query = define.engine.execute("""
+        SELECT lo.login_name, jo.journalid, jo.title, jo.content
+        FROM login lo
+        INNER JOIN journal jo USING (userid)
+        WHERE jo.is_spam IS TRUE
+    """).fetchall()
+    return Response(define.webpage(
+        request.userid,
+        "modcontrol/spamqueue/journal.html",
+        [query],
+        title="Journal Spam Queue"
+    ))
+
+
+@moderator_only
+@token_checked
+def modcontrol_spamqueue_journal_post_(request):
+    """
+    A Pyramid route controller for approving for display or items which the spam filter has tagged as being spam.
+    :param request: The Pyramid request.
+    :return/raises: HTTPSeeOther for the journal spam queue.
+    """
+    directive = request.params.get("directive")
+    if directive:
+        journalid, action = directive.split(",")
+    else:
+        raise HTTPSeeOther("/modcontrol/spamqueue/journal")
+
+    if action == "approve":
+        # Approve and insert the journal into the notifications table.
+        journalid, userid, rating, settings, content, ua_id, ip_addr = define.engine.execute("""
+            UPDATE journal
+            SET is_spam = FALSE
+            WHERE journalid = %(id)s
+            RETURNING journalid, userid, rating, settings, content, submitter_user_agent_id, submitter_ip_address
+        """, id=journalid).first()
+        # Update the spam filtering backend to indicate that this was not spam
+        spam_filtering.submit(
+            is_ham=True,
+            user_ip=ip_addr,
+            user_agent_id=ua_id,
+            user_id=userid,
+            comment_type="journal",
+            comment_content=content,
+        )
+        welcome.journal_insert(userid=userid, journalid=journalid, rating=rating, settings=settings)
+    elif action == "delete":
+        # Delete and purge the journal from the welcome table
+        define.engine.execute("""
+            DELETE FROM journal
+            WHERE journalid = %(id)s AND
+                  is_spam = TRUE
+        """, id=journalid)
```

Comment:
In theory, ``journal`` records shouldn't be as touchy in terms of remaining artifacts as ``submission`` records (full size image, jpg/png/webp thumbnails), but I don't like this either.